tests/functional/test_model_completeness.py

```diff
@@ -34,15 +38,53 @@ def test_paginators_and_waiters_are_not_lost_in_new_version(
 ):
     # Make sure if a paginator and/or waiter exists in previous version,
     # there will be a successor existing in latest version.
-    loader = Loader()
     try:
-        loader.load_service_model(service_name, type_name, previous_version)
+        LOADER.load_service_model(service_name, type_name, previous_version)
     except DataNotFoundError:
         pass
     else:
         try:
-            loader.load_service_model(service_name, type_name, latest_version)
+            LOADER.load_service_model(service_name, type_name, latest_version)
         except DataNotFoundError as e:
             raise AssertionError(
                 f"{type_name} must exist for {service_name}: {e}"
             )
+
+
+def _endpoint_rule_set_cases():
+    for service_name in AVAILABLE_SERVICES:
+        versions = LOADER.list_api_versions(service_name, 'service-2')
+        for version in versions:
+            yield service_name, version
+
+
+@pytest.mark.parametrize(
+    "service_name, version",
+    _endpoint_rule_set_cases(),
+)
+def test_all_endpoint_rule_sets_exist(service_name, version):
+    """Tests the existence of endpoint-rule-set.json for each service
+    and verifies that content is present."""
+    type_name = 'endpoint-rule-set'
+    data = LOADER.load_service_model(service_name, type_name, version)
+    assert len(data['rules']) >= 1
+
+
+test_data_dir = os.path.join(os.path.dirname(__file__), "endpoint-rules")
+
+
+@pytest.mark.parametrize("service_name", AVAILABLE_SERVICES)
+def test_all_endpoint_tests_exist(service_name):
+    """Tests the existence of endpoint-tests.json for each service
+    and verifies that content is present."""
+    file_name = 'endpoint-tests.json'
+    endpoint_tests_file = os.path.join(test_data_dir, service_name, file_name)
+    with open(endpoint_tests_file) as f:
+        data = json.load(f)
+        assert len(data['testCases']) >= 1
+
+
+def test_partitions_exists():
+    """Tests the existence of partitions.json and verifies that content is present."""
+    data = LOADER.load_data('partitions')
+    assert len(data['partitions']) >= 4
```

Comment:
do we need to explicitly check the length or can we just check for the presence of the `partitions` key?

Reply:
The length is important here since we know there will exist at least 4 partitions as Nate mentioned in [this](https://github.com/boto/botocore/pull/2747#discussion_r958674553) comment.

Reply:
got it thanks!